webapp/apps/core/views.py

```diff
@@ -0,0 +1,178 @@
+from django.shortcuts import render, get_object_or_404
+from django.http import JsonResponse
+
+from django.utils import timezone
+from .models import CoreRun
+from ..taxbrain.models import OutputUrl
+from .compute import Compute, JobFailError
+from ..formatters import get_version
+from ..taxbrain.views import TAXCALC_VERSION, WEBAPP_VERSION, dropq_compute
+from django.shortcuts import (render, render_to_response, get_object_or_404,
+                              redirect)
+from django.template.context import RequestContext
+
+
+def output_detail(request, pk):
+    """
+    This view is the single page of diplaying a progress bar for how
+    close the job is to finishing, and then it will also display the
+    job results if the job is done. Finally, it will render a 'job failed'
+    page if the job has failed.
+
+    Cases:
+        case 1: result is ready and successful
+
+        case 2: model run failed
+
+        case 3: query results
+          case 3a: all jobs have completed
+          case 3b: not all jobs have completed
+    """
+
+    try:
+        url = OutputUrl.objects.get(pk=pk)
+    except OutputUrl.DoesNotExist:
+        raise Http404
+
+    model = url.unique_inputs
+
+    taxcalc_vers_disp = get_version(url, 'taxcalc_vers', TAXCALC_VERSION)
+    webapp_vers_disp = get_version(url, 'webapp_vers', WEBAPP_VERSION)
+
+    context_vers_disp = {'taxcalc_version': taxcalc_vers_disp,
+                         'webapp_version': webapp_vers_disp}
+    if model.tax_result:
+        # try to render table; if failure render not available page
+        try:
+            context = get_result_context(model, request, url)
+        except Exception as e:
+            print('Exception rendering pk', pk, e)
+            traceback.print_exc()
+            edit_href = '/taxbrain/edit/{}/?start_year={}'.format(
+                pk,
+                model.first_year or START_YEAR  # sometimes first_year is None
+            )
+            not_avail_context = dict(edit_href=edit_href,
+                                     **context_vers_disp)
+            return render(
+                request,
+                'taxbrain/not_avail.html',
+                not_avail_context)
```

Comment:
This logic is aimed at data with a backwards incompatible format. Ideally, we won't need to accommodate backwards incompatible data. I think it's safe to remove it. Hopefully, we won't need to add it back.